Gordon360/Services/RecIM/ActivityService.cs

```diff
@@ -215,6 +218,53 @@ public bool ActivityRegistrationClosed(int activityID)
             var activity = _context.Activity.FirstOrDefault(a => a.ID == activityID);
             return (activity.RegistrationStart < DateTime.Now) && (activity.RegistrationEnd > DateTime.Now);
         }
+
+        public async Task DeleteActivityCascade(int activityID)
+        {
+            //delete participant involvement
+            var participantActivity = _context.ParticipantActivity.Where(pa => pa.ActivityID == activityID).ToList();
+            foreach (var pa in participantActivity)
+                pa.PrivTypeID = 0;
+            await _context.SaveChangesAsync();
+
+
+            // delete teams
+            var activityTeams = _context.Team.Where(t => t.ActivityID == activityID);
+            foreach (var team in activityTeams)
+            {
+                team.StatusID = 0;
```

Comment:
I don't there there is too much benefit to having a new column for something that can be handled in status. Potentially it could be "easier" or more human readable if handled as a column where we can just do searches for .Deleted, however I could just do that for .____ID == 0